pkg/oc/bootstrap/clusterup/components/registry/registry_install.go

```diff
@@ -0,0 +1,110 @@
+package registry
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/golang/glog"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/origin/pkg/oc/bootstrap/clusterup/componentinstall"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/openshift"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/run"
+	"github.com/openshift/origin/pkg/oc/errors"
+	securityclient "github.com/openshift/origin/pkg/security/generated/internalclientset/typed/security/internalversion"
+)
+
+const (
+	DefaultNamespace  = "default"
+	SvcDockerRegistry = "docker-registry"
+	masterConfigDir   = "/var/lib/origin/openshift.local.config/master"
+	RegistryServiceIP = "172.30.1.1"
+)
+
+type RegistryComponentOptions struct {
```

Comment:
Are you thinking about creating an interface that all the 'install plugins' will re-use?

Reply:
> Are you thinking about creating an interface that all the 'install plugins' will re-use?

Yes, but I want to see what I need before I commit.  I think I will end up passing more information into the install method.

Reply:
This is a lot of code for something that will eventually be using an operator.  What’s the short term benefit to formalizing this?

Reply:
> This is a lot of code for something that will eventually be using an operator. What’s the short term benefit to formalizing this?

I get to formalize input and output interfaces for different components that want to participate in cluster up.  I also get  create `oc cluster add/remove <component>` and that lets me bound the feature creep of cluster up.

Reply:
> This is a lot of code for something that will eventually be using an operator. What’s the short term benefit to formalizing this?

Also, move of this code is a move with more explicit input to help us see the true dependency structures we need to snip.

Reply:
agree, I think this formalization will help is see the dependency chain better and we can promote this to operators when we will be ready... it also helps clean up the cluster up code we have today.